src/main/java/org/example/utils/GA4RequestVerifier.java

```diff
@@ -1,4 +1,69 @@
 package org.example.utils;
 
+import net.lightbody.bmp.BrowserMobProxy;
+import net.lightbody.bmp.BrowserMobProxyServer;
+import net.lightbody.bmp.core.har.Har;
+import net.lightbody.bmp.core.har.HarEntry;
+import net.lightbody.bmp.proxy.CaptureType;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Proxy;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+
 public class GA4RequestVerifier {
+
+    private BrowserMobProxy proxy;
+    private WebDriver driver;
+
+    public void startProxyAndDriver() {
```

Comment:
You manually start and configure BrowserMobProxyServer, which is valid but not recommended in a Carina-based project.
Instead, use Carina’s built-in proxy support:
R.CONFIG.put("browserup_proxy", "true", true);
R.CONFIG.put("proxy_type", "DYNAMIC", true);
R.CONFIG.put("proxy_port", "0", true);
This allows Carina to manage proxy lifecycle automatically and reduces boilerplate code.